packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2314,6 +2316,46 @@ export class ContainerRuntime
 		return result;
 	}
 
+	/**
+	 * Returns an aliased data store with the given alias.
+	 * @param alias - The alias for the data store.
+	 * @returns - The data store as a FluidObject if it exists and is aliased. Returns undefined if the data
+	 * store doesn't exists or is not aliased.
+	 */
+	public async getAliasedDataStore(alias: string): Promise<FluidObject | undefined> {
+		await this.dataStores.waitIfPendingAlias(alias);
+		const internalId = this.internalId(alias);
+		let context: FluidDataStoreContext;
+
+		try {
+			context = await this.dataStores.getDataStore(internalId, { wait: false });
+		} catch (error) {
+			// If getting the data store fails with 404 code, the data store doesn't exist. Return undefined.
+			// Otherwise, its an unexpected failure. Throw the error.
+			if ((error as IResponseException)?.code === 404) {
+				return undefined;
+			}
+			throw error;
+		}
+
+		// If the data store is not an alias, return undefined.
+		if (!(await context.isRoot())) {
+			return undefined;
+		}
+
+		const channel = await context.realize();
+		if (channel.entryPoint !== undefined) {
+			return channel.entryPoint.get();
```

Comment:
This is a slight change in behavior from `getRootDataStore`. Here, the data store is explicitly requested which will result in calling the `initialization` functions in `PureDataObject`.
This should not be a big deal because returning the entry point here would be similar. The caller would call `get` on it as its not very useful unless `get` is called on it.

Reply:
@anthony-murphy @ChumpChief @markfields FYI

Reply:
have you consider just returning a handle here, rather than the actual object? I believe the preference is that consumers would store the handle, even to aliased things.

Reply:
so just `return channel.entryPoint`

Reply:
Yes, I considered that. The reason I chose to not return handle is because the API name has `get` in it and then users will have to call `get` on the returned object to actually get the data store. This also seems to be the theme with `entryPoint` vs `getEntryPoint` - `entryPoint` returns handle and `getEntryPoint` returns the underlying object.

Reply:
i view the distinction differently. we do getEntryPoint where we don't have a handle, otherwise we return the handle, since we have the handle here, it think it makes sense to return it.

Reply:
My thinking is that for api that operate within the container runtime we should bias toward handles to encourage their use. i'd worry that an api like this might lead a developer to alias all their data stores, store alias, and use this method to access them. we can't really stop that, but we can at least put a handle in their path which could help push a developer like this to better patterns.

Reply:
we could call the method like `getAliasedDataStoreHandle` to make it very clear

Reply:
Good point. Changed it to return a handle now. I am not sure about the name.
@markfields had suggested `getAliasedDataStoreEntryPoint` which I believe is more appropriate than using handle but the name becomes too long. Maybe `getAliasedDataStore` is good enough?

Reply:
Yeah thinking it through I still like `getAliasedDataStoreEntryPoint`

Reply:
I'm imagining a developer coming to this API that says it's for getting a "DataStore", and then you get a Handle to an anonymous FluidObject - it's not clear from the name or types what the target of that handle is.  But naming it to explicitly say it's going to return the DataStore's entryPoint makes it clear - entryPoint is (will be) a well-understood concept and they'll know exactly what the handle is to.

Reply:
Or to go a different direction to make it shorted - `accessAliasedDataStore` returns the entryPoint makes some sense, and would probably lead people to read the comment to see what "access" means :P  I like the longer name better, just throwing an idea out